lib/streamlit/elements/widgets/button_group.py

```diff
@@ -300,76 +379,161 @@ def feedback(
+            style="borderless",
         )
         return sentiment.value
 
+    @gather_metrics("pills")
+    def pills(
+        self,
+        label: str,
+        options: OptionSequence[V],
+        *,
+        selection_mode: Literal["single", "multi"] = "single",
+        default: Sequence[V] | V | None = None,
+        format_func: Callable[[Any], str] | None = None,
+        key: Key | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        disabled: bool = False,
+        label_visibility: LabelVisibility = "visible",
+    ) -> list[V] | V | None:
+        maybe_raise_label_warnings(label, label_visibility)
+
+        def _transformed_format_func(option: V) -> ButtonGroupProto.Option:
+            """If option starts with a material icon or an emoji, we extract it to send
+            it parsed to the frontend."""
+            transformed = format_func(option) if format_func else str(option)
+            transformed_parts = transformed.split(" ")
+            icon: str | None = None
+            if len(transformed_parts) > 0:
+                maybe_icon = transformed_parts[0].strip()
+                try:
+                    icon = validate_icon_or_emoji(maybe_icon)
+                    # reassamble the option string without the icon - also
+                    # works if len(transformed_parts) == 1
+                    transformed = " ".join(transformed_parts[1:])
+                except StreamlitAPIException:
+                    # we don't have a valid icon or emoji, so we just pass
+                    pass
+            return ButtonGroupProto.Option(
+                content=transformed,
+                content_icon=icon,
+            )
+
+        indexable_options = convert_to_sequence_and_check_comparable(options)
+        default_values = get_default_indices(indexable_options, default)
+
+        serde: SingleOrMultiSelectSerde[V] = SingleOrMultiSelectSerde[V](
+            indexable_options, default_values, selection_mode
+        )
+        res = self._button_group(
+            indexable_options,
+            key=key,
+            default=default_values,
+            selection_mode=selection_mode,
+            disabled=disabled,
+            format_func=_transformed_format_func,
+            serializer=serde.serialize,
+            deserializer=serde.deserialize,
+            on_change=on_change,
+            args=args,
+            kwargs=kwargs,
+            style="pills",
+            label=label,
+            label_visibility=label_visibility,
+            help=help,
+        )
+
+        if selection_mode == "multi":
+            return res.value
+
+        return res.value if selection_mode == "single" else None
+
     # Disable this more generic widget for now
     # @gather_metrics("button_group")
+    @gather_metrics("button_group")
```

Comment:
nit: since its still "internal", maybe call this ` @gather_metrics("_internal_button_group")` instead. Once we add button group officially, its better to not have metrics already from the internal command (which might have a different API).